src/main/java/com/example/Library/services/UserDetailsServiceImpl.java

```diff
@@ -25,8 +28,13 @@ public UserDetailsServiceImpl(UserRepository userRepository) {
 
     @Override
     public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
-        User user = userRepository.findByEmail(email).orElseThrow(() ->
-                new UsernameNotFoundException("User doesn't exists"));
-        return SecurityUser.fromUser(user);
+        Optional<User> user = userRepository.findByEmail(email);
+        if (user.isPresent()) {
+            return SecurityUser.fromUser(user.get());
+        } else {
+            log.error("User doesn't exists",new UsernameNotFoundException("User doesn't exists"));
```

Comment:
there is no need to create a new exception object during logging method call, simple String error message could be logged.